### PR TITLE
Update model_base.py

### PR DESCRIPTION
**Pull request by pkohvaei**
seed(args.random_state) makes the model reproducible.

class_weight='auto'  changed to 'balanced' to prevent deprecation warnings.

### CODE REVIEW

File: eden/model_base.py [hunk 1]
```diff
@@ -90,7 +91,7 @@ def vectorizer_init(self, args):
 
     def estimator_init(self, args):
         """Setup the estimator and set of matching parameter choices."""
-        estimator = SGDClassifier(average=True, class_weight='auto', shuffle=True)
+        estimator = SGDClassifier(average=True, class_weight='balanced', shuffle=True)
```
Comment (bgruening) [inline eden/model_base.py]: Is this change really needed?

Comment (pkohvaei) [inline eden/model_base.py]: Right now the script is throwing a long warning message for each new instance. Somehow 5*20 warnings per each deafult setup run. So it's up to us to decide..

Comment (fabriziocosta) [inline eden/model_base.py]: yes, yes, it has been changed throughout but in some places it did still escape

Comment (bgruening) [inline eden/model_base.py]: More tests, tests, tests ....
Thanks @pkohvaei!
